rpc/internal/srv/srv.go

```diff
@@ -26,6 +35,13 @@ type ListArgs struct {
 	Opts *zoekt.ListOptions
 }
 
+// We need a different type for JSON requests because the query needs to be
+// parsed.
+type JSONListArgs struct {
+	Q    string
```

Comment:
I have not downcased these (e.g. `Q` -> `q`) with JSON tags because it'd be inconsistent for these to be downcased but `zoekt.ListOptions` or `zoekt.SearchOptions` not to be.  I can of course downcase all of that if that's the right thing to do.